MarkBind/markbind
# Clean up docs & tests for variables

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #2283
- update docs to remove outdated mention of page variable
- removed the incorrect and unsupported usage of variable imports from `variables.md` file
- fix an outdated test case
- add in `variables.md` file to remove unnecessary warning about missing file during functional tests

**Anything you'd like to highlight/discuss:**
Was expecting a need to update functional tests but doesn't seem to be required due to how `variables.md` and how nunjucks is processed before generating the HTML.

**Testing instructions:**
- check that `markbind init` does not contain warning of "variable has no name"
- check that `npm run test` no longer shows any warning about "no such file" for ".../variables.md" or the variable has no name warning
  - can just check the action run (under Run npm run test): https://github.com/MarkBind/markbind/actions/runs/4835706341/jobs/8618241752
 - used to have:
![image](https://user-images.githubusercontent.com/41845017/235271460-360f6235-5ce8-4d3d-95e7-625c68f1808e.png)
 - now: should be no more



**Proposed commit message: (wrap lines at 72 characters)**
Clean up docs & tests for variables

Our docs on page variable is outdated due to feature removal. The
incorrect use of variable imports from json files within our variables.md
is problematic as it is no longer supported. A outdated test case still
exists for testing of page variable override. There are two warnings
of missing variables.md files in our functional test suite.

Keeping docs, test cases and implementation up-to-date is important.

Let's fix the docs, update the test cases, remove unnecessary warning
and ensure all variables.md do not contain the imports.

The above will fix the related issues around page variables and
variable imports.



---

**Checklist:** :ballot_box_with_check:






Breaking change description:

A newly generated site (e.g. via `markbind init`) will no longer contain the following erroneous line in `_markbind/variables.md`:
```
<variable from="variables.json" />
```
This way of importing variables from an external file within the global variables.md is not supported.

For existing sites migrating to the latest version, please check that you do not have the above line in your `_markbind/variables.md` file as it will not work as expected and will log a warning. Please remove such a line if found.
